src/app/logs-myplanet/logs-myplanet.component.html

```diff
@@ -24,5 +24,8 @@
         <planet-myplanet-table dataType="logs" [data]="planet.children"></planet-myplanet-table>
       </mat-expansion-panel>
     </ng-container>
+    <ng-container *ngIf="checkEmptyLogs()">
```

Comment:
@gyawaliamit7 This is another example of a function call in a template, something that we should avoid.  Angular is not able to perform change detection on this, so it will call the function and rerender the template far more than necessary.

Please make the condition based on a property.